ext/ManoptJuMPExt.jl

```diff
@@ -0,0 +1,387 @@
+end
+
+"""
+    Manopt.Optimizer()
+
+Creates a new optimizer object.
+
+The minimization of a function `f(X)` of an an array `X[1:n1,1:n2,...]`
+over a manifold `M` starting at `X0`, can be modeled as follows:
+```julia
+using JuMP
+@variable(model, X[i1=1:n1,i2=1:n2,...] in M, start = X0[i1,i2,...])
+@objective(model, Min, f(X))
+```
+The optimizer assumes that `M` has a `Array` shape described
+by `ManifoldsBase.representation_size`.
+"""
+mutable struct Optimizer <: MOI.AbstractOptimizer
+    # Manifold in which all the decision variables leave
+    manifold::Union{Nothing,ManifoldsBase.AbstractManifold}
+    # Description of the problem in Manopt
+    problem::Union{Nothing,Manopt.AbstractManoptProblem}
+    # State of the optimizer
+    state::Union{Nothing,Manopt.AbstractManoptSolverState}
+    # Starting value for each variable
+    variable_primal_start::Vector{Union{Nothing,Float64}}
+    # Sense of the optimization, e.g., min, max or no objective
+    sense::MOI.OptimizationSense
+    # Model used to compute gradient of the objective function with AD
+    nlp_model::MOI.Nonlinear.Model
+    # Solver parameters set with `MOI.RawOptimizerAttribute`
+    options::Dict{String,Any}
+    function Optimizer()
+        return new(
+            nothing,
+            nothing,
+            nothing,
+            Union{Nothing,Float64}[],
+            MOI.FEASIBILITY_SENSE,
+            MOI.Nonlinear.Model(),
+            Dict{String,Any}(),
+        )
+    end
+end
+
+MOI.get(::Optimizer, ::MOI.SolverVersion) = "0.4.37"
+
+function MOI.is_empty(model::Optimizer)
+    # TODO replace `isnothing(model.nlp_model.objective)`
+    #      by `MOI.is_empty` once the following is fixed
+    #      https://github.com/jump-dev/MathOptInterface.jl/issues/2302
+    return isnothing(model.manifold) &&
+           isempty(model.variable_primal_start) &&
+           isnothing(model.nlp_model.objective) &&
+           model.sense == MOI.FEASIBILITY_SENSE
+end
+
+function MOI.empty!(model::Optimizer)
+    model.manifold = nothing
+    model.problem = nothing
+    model.state = nothing
+    empty!(model.variable_primal_start)
+    model.sense = MOI.FEASIBILITY_SENSE
+    # TODO replace by `MOI.empty!` once the following is fixed
+    #      https://github.com/jump-dev/MathOptInterface.jl/issues/2302
+    model.nlp_model.objective = nothing
+    return nothing
+end
+
+"""
+    MOI.supports(::Optimizer, attr::MOI.RawOptimizerAttribute)
+
+Return a `Bool` indicating whether `attr.name` is a valid option name
+for `Manopt`.
+"""
+function MOI.supports(::Optimizer, ::MOI.RawOptimizerAttribute)
+    # FIXME It should depend on `attr.name`
+    return true
+end
+
+function MOI.get(model::Optimizer, attr::MOI.RawOptimizerAttribute)
+    if !MOI.supports(model, attr)
+        throw(MOI.UnsupportedAttribute(attr))
+    end
+    return model.options[attr.name]
+end
+
+function MOI.set(model::Optimizer, attr::MOI.RawOptimizerAttribute, value)
```

Comment:
Should this be documented maybe?

Also – I am not sure, how these functions would be used / what they are useful for – so similar to your request on “which functions on a manifold are necessary” – here maybe the reverse: Why are these implemented here and how are they used?

Reply:
Added a docstring